src/Core/Prefab/GlobeView.js

```diff
@@ -384,13 +385,24 @@ GlobeView.prototype.screenCoordsToNodeId = function screenCoordsToNodeId(mouse)
     return Math.round(unpack);
 };
 
+GlobeView.prototype.getDepthBuffer = function getDepthBuffer(x, y, width, height) {
+    const g = this.mainLoop.gfxEngine;
+    const previousRenderState = this._renderState;
+    this.changeRenderState(RendererConstant.DEPTH);
+    const buffer = g.renderViewTobuffer(this, g.fullSizeRenderTarget, x, y, width, height);
+    this.changeRenderState(previousRenderState);
+    return buffer;
+};
+
 const matrix = new THREE.Matrix4();
 const screen = new THREE.Vector2();
 const pickWorldPosition = new THREE.Vector3();
 const ray = new THREE.Ray();
 const direction = new THREE.Vector3();
 GlobeView.prototype.getPickingPositionFromDepth = function getPickingPositionFromDepth(mouse) {
-    const dim = this.mainLoop.gfxEngine.getWindowSize();
+    const viewPaused = this.mainLoop.scheduler.commandsWaitingExecutionCount() == 0 && (!this.controls || this.controls.isIdle());
```

Comment:
Couldn't you replace `the controls.isIdle` test with a something testing if the mainloop is halted (ie : mainLoop.renderingState == RENDERING_SCHEDULED)?

This way you wouldn't have to add a `isIdle` function to all controls.